File: .skills/openclaw-skills/skills/shifenghu/aicade-galaxy-skills/scripts/list_tools.py
#!/usr/bin/env python3

import json
import sys
from pathlib import Path
from typing import Dict, List


def main() -> None:
    try:
        args = parse_cli_args(sys.argv[1:])
        artifact_path = args.get("--artifact", "output/aicade-galaxy-skill.json")
        artifact = json.loads(Path(artifact_path).read_text(encoding="utf-8"))

        tools = artifact.get("tools")
        if not isinstance(tools, list):
            raise RuntimeError("Invalid artifact format.")

        summary = {
            "baseUrl": artifact.get("baseUrl"),
            "toolCount": artifact.get("toolCount"),
            "tools": [
                {
                    "name": tool.get("name"),
                    "title": tool.get("metadata", {}).get("title", ""),
                    "method": tool.get("metadata", {}).get("method", ""),
                    "path": tool.get("metadata", {}).get("path", ""),
                    "required": tool.get("inputSchema", {}).get("required", []),
                    "responseFields": tool.get("metadata", {}).get(
                        "responseFields", []
                    ),
                }
                for tool in tools
                if isinstance(tool, dict)
            ],
        }

        print(json.dumps(summary, ensure_ascii=False, indent=2))
    except Exception as exc:  # noqa: BLE001
        print(str(exc), file=sys.stderr)
        sys.exit(1)


def parse_cli_args(argv: List[str]) -> Dict[str, str]:
    if len(argv) % 2 != 0:
        raise RuntimeError("Usage: python3 scripts/list_tools.py --artifact <path>")

    result: Dict[str, str] = {}
    for index in range(0, len(argv), 2):
        key = argv[index]
        value = argv[index + 1]
        if not key.startswith("--"):
            raise RuntimeError(
                "Usage: python3 scripts/list_tools.py --artifact <path>"
            )
        result[key] = value
    return result


if __name__ == "__main__":
    main()

File: .skills/openclaw-skills/skills/shifenghu/aicade-galaxy-skills/scripts/list_tools.mjs
#!/usr/bin/env node

import { readFile } from "node:fs/promises";

async function main() {
  try {
    const args = parseCliArgs(process.argv.slice(2));
    const artifactPath =
      args.get("--artifact") ?? "output/aicade-galaxy-skill.json";
    const artifact = JSON.parse(await readFile(artifactPath, "utf8"));

    if (!artifact || typeof artifact !== "object" || !Array.isArray(artifact.tools)) {
      throw new Error("Invalid artifact format.");
    }

    const summary = {
      baseUrl: artifact.baseUrl,
      toolCount: artifact.toolCount,
      tools: artifact.tools.map((tool) => ({
        name: tool.name,
        title: tool.metadata?.title ?? "",
        method: tool.metadata?.method ?? "",
        path: tool.metadata?.path ?? "",
        required: Array.isArray(tool.inputSchema?.required)
          ? tool.inputSchema.required
          : [],
        responseFields: Array.isArray(tool.metadata?.responseFields)
          ? tool.metadata.responseFields
          : [],
      })),
    };

    console.log(JSON.stringify(summary, null, 2));
  } catch (error) {
    console.error(error instanceof Error ? error.message : String(error));
    process.exitCode = 1;
  }
}

function parseCliArgs(argv) {
  const result = new Map();
  for (let index = 0; index < argv.length; index += 2) {
    const key = argv[index];
    const value = argv[index + 1];
    if (!key?.startsWith("--") || value === undefined) {
      throw new Error(
        "Usage: node scripts/list_tools.mjs --artifact <path>",
      );
    }
    result.set(key, value);
  }
  return result;
}

await main();
